### PR TITLE
Login with email and password

### PR DESCRIPTION
**Pull request by Balinda21**
### What does this PR do?

Implement user login functionality.

### Description of Task to be completed?

1. Add user login endpoint to authenticate users.
2. Validate user credentials (email and password).
3. Check if the user is verified.
4. Generate and return a JWT token upon successful authentication.

### How should this be manually tested?

- Make a POST request to the login endpoint with valid email and password.
- Check the response for a success message and a JWT token.

### Any background context you want to provide?

Only users who are verified can log in so as to enhance the overall security, reliability, and integrity of the application.
also comments have been added to the code for clear understanding and maintenance.

### What are the relevant pivotal tracker stories

#187570865: As a user, I can log into the E-commerce application via email & password

### CODE REVIEW

**Review by kirengamartial — APPROVED** [review]
Great work on this PR! The code is clean and well-structured.

**Review by EdgeReachTech — APPROVED** [review]
good! code written well

**Review by rodriguecyber — APPROVED** [review]
good!  code wriitten well

File: src/routes/user.route.ts [hunk 1]
```diff
@@ -75,3 +77,5 @@ usersRouter.get('/users/:id', UserController.getUserById);
 usersRouter.get('/users', UserController.getAllUsers);
 usersRouter.put('/users/update//:id', UserController.updateUser);
 usersRouter.delete('/users/delete/:id', UserController.deleteUser);
+usersRouter.post('/login/user', login);
```
Comment (Ishimwe7) [inline src/routes/user.route.ts:80]: Here you can add swagger annotation

**Review by Ishimwe7 — COMMENTED** [review]
Good job @Balinda21 , your codes looks well and organized, please consider the highlighted section

**Review by DeliceLydia — APPROVED** [review]
Well done

**Pull request merged**